src/middleware/TestingHandler.ts

```diff
@@ -0,0 +1,265 @@
+		}
+
+		return responseBody;
+	}
+
+	/**
+	 * @private
+	 * creates a response object out of responseHeader and responseBody
+	 * @param {TestingHandlerOptions} testingHandlerOptions - The TestingHandlerOptions object
+	 * @param {string} requestURL - the URL for the request
+	 * @returns Response object
+	 */
+	private createResponse(testingHandlerOptions: TestingHandlerOptions, requestURL: string, requestMethod: string): Response {
+		try {
+			// creates a response Object out of responseHeader and responseBody
+			let responseBody;
+			let responseHeader;
+			let requestID: string;
+			let requestDate: Date;
+
+			this.setStatusCode(testingHandlerOptions, requestURL, requestMethod);
+			requestID = generateUUID();
+			requestDate = new Date();
+			responseHeader = this.createResponseHeaders(testingHandlerOptions.statusCode, testingHandlerOptions.statusMessage, requestID, requestDate);
+			responseBody = this.createResponseBody(testingHandlerOptions.statusCode, testingHandlerOptions.statusMessage, requestID, requestDate);
+			const init = { url: requestURL, status: testingHandlerOptions.statusCode, statusText: testingHandlerOptions.statusMessage, headers: responseHeader };
+			const response = new Response(responseBody, init);
+
+			return response;
+		} catch (error) {
+			throw error;
+		}
+	}
+
+	/**
+	 * @private
+	 * Fetches a random status code for the RANDOM mode from the predefined array
+	 * @param {string} requestMethod - the API method for the request
+	 * @returns the random status code
+	 */
+	private getRandomStatusCode(requestMethod: string): number {
+		try {
+			// returns random status code for the random method from the array present
+			const statusCodeArray: number[] = methodStatusCode[requestMethod] as number[];
+			return statusCodeArray[Math.floor(Math.random() * statusCodeArray.length)];
+		} catch (error) {
+			throw error;
+		}
+	}
+
+	/**
+	 * @private
+	 * To fetch the relative URL out of the complete URL
+	 * @param {RegExp} pattern - the regex pattern for the URL
+	 * @param {string} urlMethod - the complete URL
+	 * @returns the relative URL
+	 */
+	private getRelativeURL(pattern: RegExp, urlMethod: string): string {
+		// just helps in getting the relative URL from the complete url, just to match the url as in manual map
+		const relativeURL = urlMethod.replace(pattern, "");
+		return relativeURL;
+	}
+
+	/**
+	 * @private
+	 * To fetch the status code from the map(if needed), then returns response by calling createResponse
+	 * @param {TestingHandlerOptions} testingHandlerOptions - The TestingHandlerOptions object
+	 * @param {string} requestURL - the URL for the request
+	 * @param {string} requestMethod - the API method for the request
+	 * @returns Response object
+	 */
+	private setStatusCode(testingHandlerOptions: TestingHandlerOptions, requestURL: string, requestMethod: string) {
+		try {
+			// setting some random status code
+			testingHandlerOptions.statusMessage = "Some error happened";
```

Comment:
If we are hard coding the default error message, we can do the same inside TesingHandlerOptions constructor or may be we can set this as a default value for the TestingHandlerOption.statusMessage.